software/controller/lib/core/commfail_alarm.h

```diff
@@ -0,0 +1,38 @@
+/* Copyright 2020, RespiraWorks
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#ifndef COMMFAIL_ALARM_H
+#define COMMFAIL_ALARM_H
+
+#include <stdint.h>
+
+// This module monitors if any alarms are triggered
+class CommFailAlarm {
+public:
+  void Initialize();
+  // alarm_handler monitors/checks is there are any faults and despatches alarms
+  // accordingly.
```

Comment:
What is meant by "accordingly"?

Reply:
I would also think of
* preconditions
* postcondictions

to help define what each function wants to achieve. This will also make it easier to write tests.

Reply:
added a small description for each function being used